doc/source/user/basics.types.rst

```diff
@@ -12,6 +12,146 @@ Array types and conversions between types
 NumPy supports a much greater variety of numerical types than Python does.
 This section shows which are available, and how to modify an array's data-type.
 
+NumPy numerical types are instances of `numpy.dtype` (data-type) objects,
+each having unique characteristics.  Once you have imported NumPy using ``import
+numpy as np`` the dtypes are available as e.g. `numpy.bool`,
+`numpy.float32`, etc.
+
+Data-types can be used as functions to convert python numbers to array scalars
```

Comment:
Reading this sentence again feels super misleading. This is not a data type! E.g., one cannot indicate byte order with it.

Also, do we even want to encourage creation of scalars? I think we should go straight to talking about arrays.

Reply:
I reworded this a bit and now refer to these objects as scalar types (which they are) but don't explain exactly what that means when I introduce the term, since I think that would be distracting. I also added a cross-reference to the dtype reference docs so there's at least a hyperlink to more in-depth docs.